aten/src/ATen/BatchingUtils.cpp

```diff
@@ -0,0 +1,55 @@
+#include <ATen/BatchingUtils.h>
+#include <ATen/ATen.h>
+
+namespace at {
+
+static bool areBdimsAtFrontInOrder(BatchDimsRef bdims) {
+  for (int64_t idx = 0; idx < bdims.size(); idx++) {
+    if (bdims[idx].dim() != idx) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// NB: There is an invariant that BatchDims are stored in increasing
+// `level` order.
```

Comment:
This should just be a `See Note [...]` to https://github.com/pytorch/pytorch/pull/38424/files#diff-254a034cc0b96a370756c251b786e92cR81